modules/standard/Random.chpl

```diff
@@ -325,20 +344,21 @@ module Random {
 
   pragma "no doc"
   /* _choice branch for distribution defined by probabilities array */
-  proc _choiceProbabilities(stream, arr:[], size:?sizeType, replace, prob:?probType) throws
+  proc _choiceProbabilities(stream, X:domain, size:?sizeType, replace, prob:?probType) throws
   {
     import Search;
     import Sort;
-
-    // If stride, offset, or size don't match, we're in trouble
-    if arr.domain != prob.domain then
-      throw new owned IllegalArgumentError('choice() arrays must have equal domains');
-
+    
+    if prob.size != X.size {
+      throw new owned IllegalArgumentError('choice() domain.size must be equal to prob.size');
+    }
+    
     if prob.size == 0 then
-      throw new owned IllegalArgumentError('choice() arrays cannot be empty');
+      throw new owned IllegalArgumentError('choice() array cannot be empty');
```

Comment:
Let's stick with arg names:

```suggestion
      throw new owned IllegalArgumentError('choice() prob cannot be empty');
```

Reply:
@ben-albrecht wouldn't it be better to use `prob array` as used in other error calls?

Reply:
Yes, good point.